### DISCUSSION TITLE
Add vsixgallery.com as a marketplace source via the extension

### DISCUSSION THREAD

**Discussion by mrlacey** [VsixCommunity/Community.VisualStudio.Toolkit#15] [Ideas]
I don't know if it's possible to automatically add to the list of Additional Extension Galleries but if it is, it would be good to have the extension add an entry for vsixgallery.com (assuming one doesn't already exist.)

The aim of such a change is to increase adoption and support of this way of distributing nightly/preview builds of extensions.

![image](https://user-images.githubusercontent.com/189547/115545738-0f496c00-a29c-11eb-98d0-86d30e63423c.png)

**Comment by reduckted** [comment]
I'm not sure the APIs to do that are publicly available (probably for a good reason). [`IVsExtensionManager`](https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2013/dd884736(v=vs.120)) and [`IVsExtensionRepository`](https://docs.microsoft.com/en-us/previous-versions/visualstudio/visual-studio-2013/dd884797(v=vs.120)) are the only interfaces I can find that are publicly advertised, and they only allow querying the available extensions or installing extensions.

Alternatively, you could try writing directly to the settings store. [Settings Store Explorer](https://marketplace.visualstudio.com/items?itemName=PaulHarrington.SettingsStoreExplorer) would come in handy for finding out what needs to change.

**Reply by madskristensen** [reply]
It's possible to do in a .pkgdef file. I've seen some issue with that lately though, so I'd have to try. You can use the Settings Store Explorer to see how to construct the .pkgdef file.

This extension adds Open VSIX Gallery to the list: https://marketplace.visualstudio.com/items?itemName=MadsKristensen.VSIXGallery-nightlybuilds
